Core/Src/main.c

```diff
@@ -66,6 +67,15 @@ int main(void)
 {
   /* USER CODE BEGIN 1 */
 
+	/**when I tried declaring the constants using
+	 *
+	#define MAX_ADC = 1023
+	#define DUTY_CYCLE_RANGE = 3000
+	I was given some "syntax" errors
+	 */
+	int MAX_ADC = 1023;
+	int DUTY_CYCLE_RANGE = 3000;
+
```

Comment:
A couple of things here:
1. What you tried to declare is a macro, NOT a constant. They are very different.
2. If your original plan was to declare a macro, make sure you do it at the top of the file, there is no reason to define it in the main function, even though im pretty sure it will work.
3. Yes it makes sense that you got syntax errors for doing something like `#define MAX_ADC = 1023` because that is not how you define a macro. You do not use an equal sign because a macro is a 1 for 1 replacement for the identifier you create. So it would be `#define MAX_ADC 1023` and everywhere you put `MAX_ADC ` during the preprocessing stage of the compilation process, the preprocessor will literally replace it with 1023.
4. Please do not get discouraged by errors! Many times error messages tell you exactly what was wrong, all you have to do is search it up on the internet :) Someone most definitely has already done the same thing. This goes for basically any compiler error.
5. I know you only creating the ints because you couldn't define the macro, but in general, it is always best to use the data type that best represents the number. So in this case, using a uint16_t for both would be best because both numbers are less than 2^16 and they are positive!
6. Also, since these values never change, they should have been marked with const (you got the ALL_CAPS notation correct though!)